changelog/fragments/remove-test-framework.yaml

```
@@ -0,0 +1,7 @@
+entries:
+  - description: Remove `test` subcommand, the test framework `pkg/test`, and legacy Go e2e code.
```

Nit: Removing `legacy Go e2e code` is more of an internal change to our own e2e tests right? In which case we normally wouldn't document that in the CHANGELOG.
```suggestion
  - description: Remove `test` subcommand, the test framework `pkg/test`.
```

---

Yeah I was initially thinking that it'd be nice to have a reference for exactly what was removed since we're aggregating later, but now I agree with you. Removing.